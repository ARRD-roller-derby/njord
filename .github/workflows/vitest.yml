name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30