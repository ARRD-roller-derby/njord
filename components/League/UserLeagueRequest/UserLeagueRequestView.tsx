import FullscreenModale from '../../_ui/FullscreenModale/FullscreenModale'
import UserLeagueRequestForm from '../UserLeagueRequestForm/UserLeagueRequestForm'

interface props {
  readonly open: boolean
  readonly openModale: Function
  readonly closeModale: Function
  readonly isAdmin:boolean
  readonly value?:string
}
export default function UserLeagueRequestView({
  open,
  openModale,
  closeModale,
  isAdmin,
  value
}: props) {
  return (isAdmin &&
    <>
      <button onClick={() => openModale()}>Inviter</button>
      {open && (
        <FullscreenModale setClose={() => closeModale()}>
          <UserLeagueRequestForm close={closeModale} defaultValue={value}/>
        </FullscreenModale>
      )}
    </>
  )
}
