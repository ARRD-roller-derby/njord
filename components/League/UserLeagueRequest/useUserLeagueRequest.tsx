import { useState } from 'react'
import { useSession } from 'next-auth/react';

export default function useUserLeagueRequest(): {
  open: boolean
  isAdmin:boolean
  openModale: Function
  closeModale: Function
} {
  const 
  {data:session} = useSession(),
  [open, setOpen] = useState<boolean>(false)


  return {
    open,
    isAdmin:session ? !!session.isAdmin || session.user.profiles.find((o:string)=>o ==='bureau'):false,
    openModale: () => setOpen(true),
    closeModale: () => setOpen(false),
  }
}
