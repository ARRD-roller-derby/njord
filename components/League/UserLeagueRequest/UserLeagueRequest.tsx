import UserLeagueRequestView from "./UserLeagueRequestView";
import useLeagueRequest from './useUserLeagueRequest';

export default function UserLeagueRequest(){
  const props = useLeagueRequest();
  return <UserLeagueRequestView {...props}/>
}