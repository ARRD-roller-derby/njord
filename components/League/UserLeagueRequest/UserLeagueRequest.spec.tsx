import { afterEach, describe, expect, it } from 'vitest'
import { render, cleanup} from '@testing-library/react'
import UserLeagueRequest from './UserLeagueRequest'

describe('<UserLeagueRequest />', () => {
  afterEach(cleanup)

  it('Check snapshot', () => {
    const { asFragment } = render(
      <UserLeagueRequest />
    )
    expect(asFragment()).toMatchSnapshot()
  })

})
