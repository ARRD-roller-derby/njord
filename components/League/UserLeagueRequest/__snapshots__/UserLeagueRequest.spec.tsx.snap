// Vitest Snapshot v1

exports[`<UserLeagueRequest /> > Check snapshot 1`] = `<DocumentFragment />`;
